platform-sdk/swirlds-jasperdb/src/main/java/com/swirlds/merkledb/collections/LongListDisk.java

```diff
@@ -40,29 +90,103 @@ public class LongListDisk extends LongList {
      * @throws IOException If there was a problem reading the file
      */
     public LongListDisk(final Path file) throws IOException {
-        super(FileChannel.open(file, StandardOpenOption.CREATE, StandardOpenOption.READ, StandardOpenOption.WRITE));
-        this.file = file;
+        this(file, DEFAULT_RESERVED_BUFFER_LENGTH);
+    }
+
+    LongListDisk(final Path file, final long reservedBufferLength) throws IOException {
+        super(file, reservedBufferLength);
+        freeChunks = new ConcurrentLinkedDeque<>();
+    }
+
+    @Override
+    protected void readBodyFromFileChannelOnInit(String sourceFileName, FileChannel fileChannel) throws IOException {
+        Path tempFile = createTempFile(sourceFileName);
+        // create temporary file for writing
+        @SuppressWarnings("resource")
```

Comment:
You're suppressing the warning here, but it looks like it's valid.  What are you doing to ensure this RandomAccessFile is closed?

Reply:
Initially I was closing it but then quickly realized that the file channel I take from it gets closed as well, and it's not supposed to be closed because this implementation use it to write values to the disk. I'll add the comment there.

Reply:
It's fine to use the channel, but somewhere the file needs to be closed, otherwise we leak handles.  Does it get closed somewhere?

Reply:
I refactored the code  - now it closes `RandomAccessFile` (and it's underlying channel) and create another channel to be used to access this file.